src/core/lib/promise/detail/basic_seq.h

```diff
@@ -17,384 +17,13 @@
 
 #include <grpc/support/port_platform.h>
 
-#include <array>
-#include <cassert>
-#include <new>
-#include <tuple>
-#include <utility>
-
-#include "absl/meta/type_traits.h"
-#include "absl/utility/utility.h"
-
 #include "src/core/lib/gprpp/construct_destruct.h"
-#include "src/core/lib/promise/detail/promise_factory.h"
-#include "src/core/lib/promise/detail/promise_like.h"
-#include "src/core/lib/promise/detail/switch.h"
 #include "src/core/lib/promise/poll.h"
 
 namespace grpc_core {
 namespace promise_detail {
 
-// Helper for SeqState to evaluate some common types to all partial
-// specializations.
-template <template <typename> class Traits, typename FPromise, typename FNext>
-struct SeqStateTypes {
-  // Our current promise.
-  using Promise = FPromise;
-  // The result of our current promise.
-  using PromiseResult = typename Promise::Result;
-  // Traits around the result of our promise.
-  using PromiseResultTraits = Traits<PromiseResult>;
-  // Wrap the factory callable in our factory wrapper to deal with common edge
-  // cases. We use the 'unwrapped type' from the traits, so for instance, TrySeq
-  // can pass back a T from a StatusOr<T>.
-  using Next = promise_detail::OncePromiseFactory<
-      typename PromiseResultTraits::UnwrappedType, FNext>;
-};
-
-// One state in a sequence.
```

Comment:
We should move the comments from this file into the generated one, for readability.